core/iwasm/libraries/lib-wasi-threads/test/update_shared_data_and_alloc_heap.c

```diff
@@ -60,6 +57,11 @@ main(int argc, char **argv)
     assert(count != NULL && "Failed to call calloc");
     assert(pthread_mutex_init(&mutex, NULL) == 0 && "Failed to init mutex");
 
+    for (int i = 0; i < NUM_THREADS; i++) {
+        vals[i] = malloc(sizeof(int *));
```

Comment:
```suggestion
        vals[i] = malloc(sizeof(int));
```

Reply:
The point of this part of the test was to try heap allocation from the spawned thread, that's why I was putting it in `__wasi_thread_start_C`

Reply:
The issue is that `malloc` function is not only called here, but also called in __wasi_thread_spawn when allocating aux stack:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/libraries/thread-mgr/thread_manager.c#L147

Two threads may call malloc simultaneously.

If the case isn't modified, `out of bounds memory access` exception is often thrown.